ios/Classes/SwiftMparticleFlutterSdkPlugin.swift

```diff
@@ -234,6 +234,114 @@ public class SwiftMparticleFlutterSdkPlugin: NSObject, FlutterPlugin {
            let user = MParticle.sharedInstance().identity.getUser(NSNumber(value:mpid)) {
             user.setUserTag(attributeKey)
         }
+    case "getGDPRConsentState":
+        if let callArguments = call.arguments as? [String: Any],
+           let mpidString = callArguments["mpid"] as? String,
+           let mpid = Int(mpidString),
+           let user = MParticle.sharedInstance().identity.getUser(NSNumber(value:mpid)) {
+            var gdprConsentDictionary = [String: Any]()
+            if let gdprConsent = user.consentState()?.gdprConsentState() {
+                for purpose in gdprConsent.keys {
+                    let consentObject = gdprConsent[purpose]
+                    var consentDictionary = [String: Any]()
+                    consentDictionary["consented"] = consentObject?.consented
+                    consentDictionary["document"] = consentObject?.document
+                    consentDictionary["timestamp"] = consentObject?.timestamp.timeIntervalSince1970
+                    consentDictionary["location"] = consentObject?.location
+                    consentDictionary["hardwareId"] = consentObject?.hardwareId
+                    gdprConsentDictionary[purpose] = consentDictionary
+                }
+            }
+            result(asStringForStringKey(jsonDictionary: gdprConsentDictionary))
+        } else {
+            result("")
+        }
+    case "addGDPRConsentState":
+        if let callArguments = call.arguments as? [String: Any],
+           let consented = callArguments["consented"] as? Bool,
+           let purpose = callArguments["purpose"] as? String,
+           let mpidString = callArguments["mpid"] as? String,
+           let mpid = Int64(mpidString),
+           let user = MParticle.sharedInstance().identity.getUser(NSNumber(value:mpid)) {
+               let consentGDPR = MPGDPRConsent()
+               consentGDPR.consented = consented
+               consentGDPR.document = callArguments["document"] as? String
+               if let timestampNumber = callArguments["timestamp"] as? NSNumber {
+                   consentGDPR.timestamp = Date(timeIntervalSince1970: timestampNumber.doubleValue/1000)
+               }
+               consentGDPR.location = callArguments["location"] as? String
+               consentGDPR.hardwareId = callArguments["hardwareId"] as? String
+
+               let consentState = MPConsentState()
+               consentState.addGDPRConsentState(consentGDPR, purpose: purpose)
+               if let ccpa = user.consentState()?.ccpaConsentState() {
+                   consentState.setCCPA(ccpa)
+               }
```

Comment:
actually i was looking at the code again and it looks like you swap CCPA here for GDPR